lib/devise/models/authenticatable.rb

```diff
@@ -103,7 +103,7 @@ def authenticatable_salt
       # and passing a new list of attributes you want to exempt. All attributes
       # given to :except will simply add names to exempt to Devise internal list.
       def serializable_hash(options = nil)
-        options = options.try(:dup) || {}
+        options = options.try(:deep_dup) || {}
```

Comment:
Could we use your alternative suggestion, `dup` on the line below? I think it makes it more explicit about what we're duping / touching, and it will prevent duping other objects that may be passed in that'd be duped with `deep_dup`.

You can also update this test: https://github.com/heartcombo/devise/blob/f26e05c20079c9acded3c0ee16da0df435a28997/test/models/serializable_test.rb#L45-L47, maybe something like this to catch this scenario as well:

```ruby
assert_key "username", @user.as_json({ only: :username, except: [:email].freeze }.freeze)["user"]
```

Reply:
done and done - both solid suggestions, thanks!